src/cpu/o3/free_list.cc

```
@@ -39,14 +44,156 @@ namespace o3
+unsigned
+UnifiedFreeList::donorQuota(RegClassType type) const
+{
+    return std::min(base(type), donorReserveRegs);
+}
+
+unsigned
+UnifiedFreeList::borrowingLimit(RegClassType type, ThreadID tid) const
+{
+    unsigned reserved = 0;
+    for (ThreadID other = 0; other < numThreads; ++other) {
+        if (other == tid) {
```

Cppcheck (2.21.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)